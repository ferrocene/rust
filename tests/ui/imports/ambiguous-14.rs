//@ revisions: edition2015 edition2021
//@[edition2015] edition: 2015
//@[edition2015] check-pass
//@[edition2021] edition: 2021
// https://github.com/rust-lang/rust/issues/98467

mod a {
    pub fn foo() {}
}

mod b {
    pub fn foo() {}
}

mod f {
    pub use a::*;
    //[edition2021]~^ E0432
    pub use b::*;
    //[edition2021]~^ E0432
}

mod g {
    pub use a::*;
    //[edition2021]~^ E0432
    pub use f::*;
    //[edition2021]~^ E0432
}

fn main() {
    g::foo();
    //[edition2015]~^ WARNING `foo` is ambiguous
    //[edition2015]~| WARNING this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
}
