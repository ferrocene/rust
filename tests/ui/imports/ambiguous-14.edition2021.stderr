error[E0432]: unresolved import `a`
  --> $DIR/ambiguous-14.rs:16:13
   |
LL |     pub use a::*;
   |             ^ help: a similar path exists: `crate::a`
   |
   = note: `use` statements changed in Rust 2018; read more at <https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html>

error[E0432]: unresolved import `b`
  --> $DIR/ambiguous-14.rs:18:13
   |
LL |     pub use b::*;
   |             ^ help: a similar path exists: `crate::b`
   |
   = note: `use` statements changed in Rust 2018; read more at <https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html>

error[E0432]: unresolved import `a`
  --> $DIR/ambiguous-14.rs:23:13
   |
LL |     pub use a::*;
   |             ^ help: a similar path exists: `crate::a`
   |
   = note: `use` statements changed in Rust 2018; read more at <https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html>

error[E0432]: unresolved import `f`
  --> $DIR/ambiguous-14.rs:25:13
   |
LL |     pub use f::*;
   |             ^ help: a similar path exists: `crate::f`
   |
   = note: `use` statements changed in Rust 2018; read more at <https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html>

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0432`.
